app/views/papers/_tablist.html.erb

```diff
@@ -1,14 +1,16 @@
 <% paper = reference.paper %>
-<ul class="nav nav-pills" role="tablist">
-  <% %i{abstract details comments}.each do |tab| %>
-    <li><%= link_to tab.capitalize, '', "data-detail-id" => "#{tab}-#{tab_id_for(paper)}", class: "small paper-detail-tab" %></li>
+<% comments_count = reference.comments.map {|c| c.self_and_descendants.length}.inject(&:+) %>
+<ul class="nav nav-pills small" role="tablist">
+  <% %i{abstract details}.each do |tab| %>
+    <li><%= link_to tab.capitalize, '', "data-detail-id" => "#{tab}-#{tab_id_for(paper)}", class: "paper-detail-tab" %></li>
   <% end %>
+  <li><%= link_to comments_count ? "Comments(#{comments_count})" : "Comments", '', "data-detail-id" => "comments-#{tab_id_for(paper)}", class: " paper-detail-tab" %></li>
```

Comment:
I think this will always evaluate to true, even if it's 0. I'm just gonna change this a bit.